solr/core/src/java/org/apache/solr/pkg/PackageListeners.java

```diff
@@ -69,7 +69,7 @@ synchronized void packagesUpdated(List<PackageLoader.Package> pkgs) {
         invokeListeners(pkgInfo);
       }
     } finally {
-      MDCLoggingContext.clear();
+      if(core != null) MDCLoggingContext.clear();
```

Comment:
No; it was correct before.  Internally, setCore can accept null and doesn't require the caller to care if core is null.  Furthermore, PackageListeners.core will never be null.  It ought to be declared final as well; it's only set in the constructor.

Reply:
> PackageListeners.core will never be null

Now it's not.

The schema plugins will no more be associated with a core. They will have it as `null`. I don't need to deal with it in this. So, I'll remove the null check now. `PackageListeners` is not used in ContainerPlugins